go/packages/slslambda/event.go

```diff
@@ -16,19 +18,37 @@ const (
 )
 
 type (
+	EventOptions struct {
+		CustomFingerprint *string
+		CustomTags        map[string]string
+	}
 	errorEvent struct {
 		timestamp time.Time
 		error
+		*EventOptions
 	}
 	warningEvent struct {
 		timestamp time.Time
 		message   string
+		*EventOptions
 	}
 	protoEvent interface {
 		ToProto(traceID, parentSpanID []byte) (*instrumentationv1.Event, error)
 	}
 )
 
+func marshalCustomTags(options *EventOptions) *string {
+	if options.CustomTags != nil && len(options.CustomTags) > 0 {
+		b, err := json.Marshal(options.CustomTags)
+		if err != nil {
+			return nil
+		}
```

Comment:
I think we should report possible error here somehow

Reply:
I think we could do that if we had an interface for internal sdk errors in the go sdk, but we don't yet. I can see if it is feasible  without too much of a rewrite.

Reply:
I mean, for now, even a simple log is ok. I'd like to avoid ignoring errors completely.

Reply:
Let's just do a debug log then. I don't want us creating noise in users logs.